tile-group.lisp

```diff
@@ -1052,6 +1052,39 @@ space."
             (show-frame-indicator group))
           (run-hook-with-args *remove-split-hook* l frame)))))
 
+(defcommand (remove-sibling tile-group)
+    (&optional (group (current-group))
+	       (frame (let ((window-type (type-of (current-window))))
+			           (if (eq window-type 'float-window)
+			               nil ;; (focus-all (current-window))
+			               (window-frame (current-window))))))
+    ()
+  "removes the closest sibling frame to the specified frame, which defaults
+to the current frame. if the closes sibling is split, we call 
+remove-split-sibling, otherwise we just remove the sibling frame. "
+  (when frame
+    (labels ((remove-split-sibling (tree group)
+	       (let ((left (car tree))
+		     (right (second tree)))
+		 (when (listp left)
+		   (remove-split-sibling left group))
+		 (when (listp right)
+		   (remove-split-sibling right group))
+		 (when (frame-p left)
+		   (remove-split group left))
+		 (when (frame-p right)
+		   (remove-split group right)))))
```

Comment:
is there a reason this isn't a cond?

Reply:
When I write this as 
`(cond ((frame-p left) (remove-split group left)) 
           ((frame-p right) (remove-split group right)))`
and try to call it on a sibling frame that has been split, it removes only the split in the sibling frame, not the sibling frame itself, wheras when its implemented using two when statements, it removes both the split in the sibling frame, and the split that has created the sibling frame. I'm sorry im having a hard time expressing this well.